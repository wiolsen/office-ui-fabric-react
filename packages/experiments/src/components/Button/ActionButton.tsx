import * as React from 'react';
import { Button } from './Button';
import { IButtonComponent, IButtonProps, IButtonTokenReturnType } from './Button.types';
import { ButtonVariantsType } from './ButtonVariants.types';
import { FontWeights } from '../../Styling';

const baseTokens: IButtonComponent['tokens'] = (props, theme): IButtonTokenReturnType => {
  const { palette, semanticColors } = theme;

  return {
    backgroundColor: 'transparent',
    backgroundColorHovered: 'transparent',
    backgroundColorPressed: 'transparent',
    borderColor: 'transparent',
    borderColorHovered: 'transparent',
    borderColorPressed: 'transparent',
    color: semanticColors.buttonText,
    colorHovered: palette.themePrimary,
    colorPressed: palette.black,
    contentPadding: '0px 8px',
    height: '40px',
    iconColor: semanticColors.buttonText,
    iconColorHovered: palette.themePrimary,
    iconColorPressed: palette.black,
    textWeight: FontWeights.regular
  };
};

const disabledTokens: IButtonComponent['tokens'] = (props, theme): IButtonTokenReturnType => {
  const { semanticColors } = theme;

  return {
    color: semanticColors.buttonTextDisabled,
    colorHovered: semanticColors.buttonTextDisabled,
    colorPressed: semanticColors.buttonTextDisabled,
    iconColor: semanticColors.buttonTextDisabled,
    iconColorHovered: semanticColors.buttonTextDisabled,
    iconColorPressed: semanticColors.buttonTextDisabled
  };
};

export const ActionButtonTokens: IButtonComponent['tokens'] = (props, theme): IButtonTokenReturnType => [
  baseTokens,
  props.disabled && disabledTokens
];

const ActionButtonStackProps: IButtonProps['stack'] = {
  horizontalAlign: 'start'
};

export const ActionButton: ButtonVariantsType = props => {
  const { text, iconProps, ...rest } = props;

  return <Button stack={ActionButtonStackProps} content={text} icon={iconProps} tokens={ActionButtonTokens} {...rest} />;
};
