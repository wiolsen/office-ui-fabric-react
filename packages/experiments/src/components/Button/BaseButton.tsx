import * as React from 'react';
import { Button } from './Button';
import { IButtonComponent, IButtonStylesReturnType, IButtonTokenReturnType } from './Button.types';
import { ButtonVariantsType } from './ButtonVariants.types';

const baseTokens: IButtonComponent['tokens'] = {
  backgroundColor: undefined,
  backgroundColorHovered: undefined,
  backgroundColorPressed: undefined,
  borderColor: undefined,
  borderColorFocused: undefined,
  borderColorHovered: undefined,
  borderColorPressed: undefined,
  borderRadius: undefined,
  borderWidth: undefined,
  color: undefined,
  colorHovered: undefined,
  colorPressed: undefined,
  contentPadding: '1px 6px',
  contentPaddingFocused: undefined,
  cursor: undefined,
  height: undefined,
  iconColor: undefined,
  iconColorHovered: undefined,
  iconColorPressed: undefined,
  iconSize: undefined,
  iconWeight: undefined,
  lineHeight: undefined,
  minHeight: undefined,
  minWidth: undefined,
  textFamily: undefined,
  textSize: 13.333,
  textWeight: undefined,
  width: undefined
};

const disabledTokens: IButtonComponent['tokens'] = {
  color: 'grayText',
  colorHovered: 'grayText',
  colorPressed: 'grayText'
};

const BaseButtonTokens: IButtonComponent['tokens'] = (props, theme): IButtonTokenReturnType => [
  baseTokens,
  props.disabled && disabledTokens
];

const BaseButtonStyles: IButtonComponent['styles'] = (props, theme, tokens): IButtonStylesReturnType => {
  const { disabled } = props;

  return {
    root: {
      borderStyle: 'outset',
      overflow: 'visible',

      selectors: {
        ':active': {
          borderStyle: disabled ? 'outset' : 'inset',
          outline: 'none'
        }
      }
    }
  };
};

export const BaseButton: ButtonVariantsType = props => {
  const { text, iconProps, ...rest } = props;

  return <Button content={text} icon={iconProps} tokens={BaseButtonTokens} styles={BaseButtonStyles} {...rest} />;
};
