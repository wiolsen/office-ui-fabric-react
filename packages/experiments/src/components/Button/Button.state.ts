import { useImperativeHandle, useRef } from 'react';
import { IButtonComponent, IButtonViewProps } from './Button.types';

export const useButtonState: IButtonComponent['state'] = props => {
  const buttonRef = useRef<HTMLButtonElement | null>(null);

  useImperativeHandle(props.componentRef, () => ({
    focus: () => {
      buttonRef.current && buttonRef.current.focus();
    }
  }));

  const viewProps: IButtonViewProps = {
    ...props,
    buttonRef
  };

  return viewProps;
};
