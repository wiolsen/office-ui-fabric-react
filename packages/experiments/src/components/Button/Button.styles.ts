import { IButtonComponent, IButtonStylesReturnType, IButtonTokenReturnType } from './Button.types';
import { getFocusStyle, getGlobalClassNames, FontWeights, HighContrastSelector } from '../../Styling';
import { IsFocusVisibleClassName } from '../../Utilities';

const baseTokens: IButtonComponent['tokens'] = (props, theme): IButtonTokenReturnType => {
  const { effects } = theme;

  return {
    borderRadius: effects.roundedCorner2,
    borderStyle: 'solid',
    borderStyleFocused: 'solid',
    borderWidth: 1,
    cursor: 'pointer',
    minWidth: 100,
    minHeight: 32,
    lineHeight: 1,
    contentPadding: '0px 20px',
    textFamily: 'inherit',
    textSize: 14,
    textWeight: FontWeights.semibold,
    iconSize: 16,
    iconWeight: 400
  };
};

const hrefTokens: IButtonComponent['tokens'] = {
  contentPadding: '8px 16px'
};

const circularTokens: IButtonComponent['tokens'] = (props, theme): IButtonTokenReturnType => {
  const { semanticColors } = theme;

  return {
    borderRadius: '50%',
    borderStyleFocused: 'double',
    borderWidthFocused: 3,
    minWidth: 32,
    minHeight: 32,
    contentPadding: '',
    borderColorFocused: semanticColors.focusBorder,
    outlineColor: 'transparent',
    contentPaddingFocused: 1,
    backgroundClip: 'content-box'
  };
};

const enabledTokens: IButtonComponent['tokens'] = (props, theme): IButtonTokenReturnType => {
  const { palette, semanticColors } = theme;
  return {
    backgroundColor: palette.white,
    backgroundColorHovered: palette.neutralLighter,
    backgroundColorPressed: semanticColors.buttonBackgroundPressed,

    iconColor: semanticColors.buttonText,
    iconColorHovered: semanticColors.buttonTextHovered,
    iconColorPressed: semanticColors.buttonTextPressed,

    highContrastIconColorHovered: 'Highlight',
    highContrastIconColorPressed: 'Highlight',

    color: semanticColors.buttonText,
    colorHovered: semanticColors.buttonTextHovered,
    colorPressed: semanticColors.buttonTextPressed,

    highContrastColorHovered: 'Highlight',
    highContrastColorPressed: 'Highlight',

    borderColor: semanticColors.buttonBorder,
    borderColorHovered: semanticColors.buttonBorder,
    borderColorPressed: semanticColors.buttonBorder,

    highContrastBorderColorHovered: 'Highlight',
    highContrastBorderColorPressed: 'Highlight'
  };
};

const disabledTokens: IButtonComponent['tokens'] = (props, theme): IButtonTokenReturnType => {
  const { semanticColors } = theme;
  return {
    backgroundColor: semanticColors.buttonBackgroundDisabled,
    backgroundColorHovered: semanticColors.buttonBackgroundDisabled,
    backgroundColorPressed: semanticColors.buttonBackgroundDisabled,

    highContrastBackgroundColor: 'Window',
    highContrastBackgroundColorHovered: 'Window',
    highContrastBackgroundColorPressed: 'Window',

    iconColor: semanticColors.buttonTextDisabled,
    iconColorHovered: semanticColors.buttonTextDisabled,
    iconColorPressed: semanticColors.buttonTextDisabled,

    highContrastIconColor: 'GrayText',
    highContrastIconColorHovered: 'GrayText',
    highContrastIconColorPressed: 'GrayText',

    color: semanticColors.buttonTextDisabled,
    colorHovered: semanticColors.buttonTextDisabled,
    colorPressed: semanticColors.buttonTextDisabled,

    highContrastColor: 'GrayText',
    highContrastColorHovered: 'GrayText',
    highContrastColorPressed: 'GrayText',

    borderColor: semanticColors.buttonBorderDisabled,
    borderColorHovered: semanticColors.buttonBorderDisabled,
    borderColorPressed: semanticColors.buttonBorderDisabled,

    highContrastBorderColor: 'GrayText',
    highContrastBorderColorHovered: 'GrayText',
    highContrastBorderColorPressed: 'GrayText',

    cursor: 'default'
  };
};

const primaryEnabledTokens: IButtonComponent['tokens'] = (props, theme): IButtonTokenReturnType => {
  const { semanticColors } = theme;
  return {
    backgroundColor: semanticColors.primaryButtonBackground,
    backgroundColorHovered: semanticColors.primaryButtonBackgroundHovered,
    backgroundColorPressed: semanticColors.primaryButtonBackgroundPressed,

    highContrastBackgroundColor: 'WindowText',
    highContrastBackgroundColorHovered: 'Highlight',
    highContrastBackgroundColorPressed: 'Highlight',

    color: semanticColors.primaryButtonText,
    colorHovered: semanticColors.primaryButtonTextHovered,
    colorPressed: semanticColors.primaryButtonTextPressed,

    highContrastColor: 'Window',
    highContrastColorHovered: 'Window',
    highContrastColorPressed: 'Window',

    iconColor: semanticColors.primaryButtonText,
    iconColorHovered: semanticColors.primaryButtonTextHovered,
    iconColorPressed: semanticColors.primaryButtonTextPressed,

    highContrastIconColor: 'Window',
    highContrastIconColorHovered: 'Window',
    highContrastIconColorPressed: 'Window',

    borderWidth: 0,

    outlineColor: 'transparent',

    msHighContrastAdjust: 'none'
  };
};

const primaryCircularTokens: IButtonComponent['tokens'] = {
  borderStyleFocused: 'solid',
  borderWidthFocused: 1
};

const checkedTokens: IButtonComponent['tokens'] = (props, theme): IButtonTokenReturnType => {
  const { semanticColors } = theme;
  return {
    backgroundColor: semanticColors.buttonBackgroundChecked,
    backgroundColorHovered: semanticColors.buttonBackgroundCheckedHovered,
    backgroundColorPressed: semanticColors.buttonBackgroundPressed,

    color: semanticColors.buttonTextChecked,
    colorHovered: semanticColors.buttonTextCheckedHovered,
    colorPressed: semanticColors.buttonTextPressed,

    iconColor: semanticColors.buttonTextChecked,
    iconColorHovered: semanticColors.buttonTextCheckedHovered,
    iconColorPressed: semanticColors.buttonTextPressed
  };
};

const primaryCheckedTokens: IButtonComponent['tokens'] = (props, theme): IButtonTokenReturnType => {
  const { semanticColors } = theme;
  return {
    backgroundColor: semanticColors.primaryButtonBackgroundPressed,
    backgroundColorHovered: semanticColors.primaryButtonBackgroundHovered,
    backgroundColorPressed: semanticColors.primaryButtonBackgroundPressed,

    color: semanticColors.primaryButtonTextPressed,
    colorHovered: semanticColors.primaryButtonTextHovered,
    colorPressed: semanticColors.primaryButtonTextPressed,

    iconColor: semanticColors.primaryButtonTextPressed,
    iconColorHovered: semanticColors.primaryButtonTextHovered,
    iconColorPressed: semanticColors.primaryButtonTextPressed
  };
};

export const ButtonTokens: IButtonComponent['tokens'] = (props, theme): IButtonTokenReturnType => [
  baseTokens,
  !!props.href && hrefTokens,
  !props.disabled && enabledTokens,
  props.primary && primaryEnabledTokens,
  props.circular && circularTokens,
  props.circular && props.primary && primaryCircularTokens,
  props.checked && checkedTokens,
  props.checked && props.primary && primaryCheckedTokens,
  props.disabled && disabledTokens
];

const GlobalClassNames = {
  msButton: 'ms-Button',
  msButtonIcon: 'ms-Button-icon'
};

export const ButtonStyles: IButtonComponent['styles'] = (props, theme, tokens): IButtonStylesReturnType => {
  const { className, circular } = props;

  const globalClassNames = getGlobalClassNames(GlobalClassNames, theme);

  return {
    root: [
      globalClassNames.msButton,
      !circular && getFocusStyle(theme, { inset: 1, outlineColor: tokens.outlineColor }),
      circular && {
        selectors: {
          [`.${IsFocusVisibleClassName} &:focus`]: {
            borderWidth: 1
          }
        }
      },
      theme.fonts.medium,
      {
        backgroundColor: tokens.backgroundColor,
        borderColor: tokens.borderColor,
        borderRadius: tokens.borderRadius,
        borderStyle: tokens.borderStyle,
        borderWidth: tokens.borderWidth,
        boxSizing: 'border-box',
        color: tokens.color,
        cursor: tokens.cursor,
        display: 'inline-block',
        fontSize: tokens.textSize,
        fontWeight: tokens.textWeight,
        height: tokens.height,
        justifyContent: 'center',
        margin: 0,
        minWidth: tokens.minWidth,
        minHeight: tokens.minHeight,
        overflow: 'hidden',
        padding: 0,
        textDecoration: 'none',
        textAlign: 'center',
        userSelect: 'none',
        verticalAlign: 'baseline',
        width: tokens.width,
        outlineColor: tokens.outlineColor,

        selectors: {
          [HighContrastSelector]: {
            backgroundColor: tokens.highContrastBackgroundColor,
            borderColor: tokens.highContrastBorderColor,
            borderWidth: 1,
            color: tokens.highContrastColor,
            MsHighContrastAdjust: tokens.msHighContrastAdjust
          },
          ':hover': {
            backgroundColor: tokens.backgroundColorHovered,
            color: tokens.colorHovered,
            borderColor: tokens.borderColorHovered,

            selectors: {
              [HighContrastSelector]: {
                backgroundColor: tokens.highContrastBackgroundColorHovered,
                color: tokens.highContrastColorHovered,
                borderColor: tokens.highContrastBorderColorHovered
              }
            }
          },
          ':active': {
            backgroundColor: tokens.backgroundColorPressed,
            color: tokens.colorPressed,
            borderColor: tokens.borderColorPressed,

            selectors: {
              [HighContrastSelector]: {
                backgroundColor: tokens.highContrastBackgroundColorPressed,
                color: tokens.highContrastColorPressed,
                borderColor: tokens.highContrastBorderColorPressed
              }
            }
          },
          [`:hover .${globalClassNames.msButtonIcon}`]: {
            color: tokens.iconColorHovered,

            selectors: {
              [HighContrastSelector]: {
                color: tokens.highContrastIconColorHovered
              }
            }
          },
          [`:active .${globalClassNames.msButtonIcon}`]: {
            color: tokens.iconColorPressed,

            selectors: {
              [HighContrastSelector]: {
                color: tokens.highContrastIconColorPressed
              }
            }
          },
          // We have this here to establish the focus style of circular Buttons. If we use getFocusStyle to get the focus style, then the
          // focus style for circular Buttons becomes busted, and the way to fix it is via the backgroundClip and padding attributes, which
          // we don't have access to via getFocusStyle.
          [`.${IsFocusVisibleClassName} &:focus`]: {
            borderColor: tokens.borderColorFocused,
            borderStyle: tokens.borderStyleFocused,
            borderWidth: tokens.borderWidthFocused,
            outlineColor: tokens.outlineColor,
            backgroundClip: tokens.backgroundClip,
            padding: tokens.contentPaddingFocused
          }
        }
      },
      className
    ],
    stack: {
      padding: tokens.contentPadding,
      height: '100%'
    },
    icon: [
      globalClassNames.msButtonIcon,
      {
        display: 'flex',
        fontSize: tokens.iconSize,
        color: tokens.iconColor,
        fill: tokens.iconColor,
        // tslint:disable-next-line:no-any
        fontWeight: tokens.iconWeight as any,

        selectors: {
          [HighContrastSelector]: {
            color: tokens.highContrastIconColor
          }
        }
      }
    ],
    content: {
      overflow: 'visible',
      fontFamily: tokens.textFamily,
      fontSize: tokens.textSize,
      fontWeight: tokens.textWeight
    }
  };
};
