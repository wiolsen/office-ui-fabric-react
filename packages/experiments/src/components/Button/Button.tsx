import { createComponent } from '../../Foundation';
import { useButtonState as state } from './Button.state';
import { ButtonStyles as styles, ButtonTokens as tokens } from './Button.styles';
import { IButtonProps } from './Button.types';
import { ButtonView } from './Button.view';

export const Button: React.StatelessComponent<IButtonProps> = createComponent(ButtonView, {
  displayName: 'Button',
  state,
  styles,
  tokens
});

export default Button;
