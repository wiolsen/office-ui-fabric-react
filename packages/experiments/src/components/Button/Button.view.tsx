/** @jsx withSlots */
import { Stack, Text, KeytipData } from 'office-ui-fabric-react';
import { withSlots, getSlots } from '../../Foundation';
import { getNativeProps, buttonProperties } from '../../Utilities';
import { Icon } from '../../utilities/factoryComponents';

import { IButtonComponent, IButtonProps, IButtonRootElements, IButtonSlots, IButtonViewProps } from './Button.types';

export const ButtonView: IButtonComponent['view'] = props => {
  const { icon, content, children, disabled, onClick, allowDisabledFocus, ariaLabel, keytipProps, buttonRef, ...rest } = props;

  // TODO: 'href' is anchor property... consider getNativeProps by root type
  const buttonProps = { ...getNativeProps(rest, buttonProperties) };

  const Slots = getSlots<IButtonProps, IButtonSlots>(props, {
    root: _deriveRootType(props),
    stack: Stack,
    icon: Icon,
    content: Text
  });

  const _onClick = (ev: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement | HTMLDivElement>) => {
    if (!disabled && onClick) {
      onClick(ev);

      if (ev.defaultPrevented) {
        return;
      }
    }
  };

  const Button = (keytipAttributes?: any): JSX.Element => (
    <Slots.root
      type="button" // stack doesn't take in native button props
      role="button"
      onClick={_onClick}
      {...buttonProps}
      {...keytipAttributes}
      disabled={disabled && !allowDisabledFocus}
      aria-disabled={disabled}
      tabIndex={!disabled || allowDisabledFocus ? 0 : undefined}
      aria-label={ariaLabel}
      ref={buttonRef}
    >
      <Slots.stack horizontal as="span" tokens={{ childrenGap: 8 }} verticalAlign="center" horizontalAlign="center" verticalFill>
        <Slots.icon />
        <Slots.content />
        {children}
      </Slots.stack>
    </Slots.root>
  );

  return keytipProps ? (
    <KeytipData keytipProps={keytipProps} disabled={disabled && !allowDisabledFocus}>
      {(keytipAttributes: any): JSX.Element => Button(keytipAttributes)}
    </KeytipData>
  ) : (
    Button()
  );
};

function _deriveRootType(props: IButtonViewProps): IButtonRootElements {
  return !!props.href ? 'a' : 'button';
}
