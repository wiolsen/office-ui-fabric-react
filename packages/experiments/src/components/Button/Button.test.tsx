import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { CommandBar, Icon, Text } from 'office-ui-fabric-react';
import { Button } from './Button';

describe('Button', () => {
  it('renders a default Button with content correctly', () => {
    const component = renderer.create(<Button content="Default button" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a disabled default Button with content correctly', () => {
    const component = renderer.create(<Button disabled content="Disabled default button" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot;
  });

  it('renders a primary Button with content correctly', () => {
    const component = renderer.create(<Button primary content="Primary button" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a disabled primary Button with content correctly', () => {
    const component = renderer.create(<Button primary disabled content="Disabled primary button" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a default Button with content and an icon correctly', () => {
    const component = renderer.create(<Button icon="Upload" content="Button with string icon" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot;
  });

  it('renders a primary Button with custom content and icon right-aligned correctly', () => {
    const component = renderer.create(
      <Button primary>
        <Text>With custom text/icon right aligned</Text>
        <Icon iconName="Upload" />
      </Button>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot;
  });

  it('renders a default circular Button with an icon correctly', () => {
    const component = renderer.create(<Button icon="PeopleAdd" circular />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot;
  });

  it('renders a disabled default circular Button with an icon correctly', () => {
    const component = renderer.create(<Button icon="Phone" circular disabled />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot;
  });

  it('renders a primary circular Button with an icon correctly', () => {
    const component = renderer.create(<Button icon="FontSize" circular primary />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot;
  });

  it('renders a disabled primary circular Button with an icon correctly', () => {
    const component = renderer.create(<Button icon="Attach" circular primary disabled />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot;
  });

  it('renders a Button inside a CommandBar correctly', () => {
    const component = renderer.create(<CommandBar items={[{ key: '0', text: 'Button 1', iconProps: { iconName: 'Upload' } }]} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot;
  });

  it('renders a default checked Button correctly', () => {
    const component = renderer.create(<Button checked content="Button" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot;
  });

  it('renders a primary checked Button correctly', () => {
    const component = renderer.create(<Button primary checked content="Button" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot;
  });

  it('renders a default circular checked Button correctly', () => {
    const component = renderer.create(<Button checked circular icon="Volume3" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot;
  });

  it('renders a primary circular checked Button correctly', () => {
    const component = renderer.create(<Button primary checked circular icon="Volume3" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot;
  });
});
