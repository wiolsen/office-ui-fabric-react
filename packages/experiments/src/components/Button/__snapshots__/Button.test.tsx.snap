// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a default Button with content correctly 1`] = `
<button
  className=
      ms-Button
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background-color: #ffffff;
        border-color: #8a8886;
        border-radius: 2px;
        border-style: solid;
        border-width: 1px;
        box-sizing: border-box;
        color: #323130;
        cursor: pointer;
        display: inline-block;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 600;
        justify-content: center;
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        min-height: 32px;
        min-width: 100px;
        outline: transparent;
        overflow: hidden;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
        position: relative;
        text-align: center;
        text-decoration: none;
        user-select: none;
        vertical-align: baseline;
      }
      &::-moz-focus-inner {
        border: 0;
      }
      .ms-Fabric--isFocusVisible &:focus:after {
        border: 1px solid #ffffff;
        bottom: 2px;
        content: "";
        left: 2px;
        outline: 1px solid #605e5c;
        position: absolute;
        right: 2px;
        top: 2px;
        z-index: 1;
      }
      @media screen and (-ms-high-contrast: active){& {
        border-width: 1px;
      }
      &:hover {
        background-color: #f3f2f1;
        border-color: #8a8886;
        color: #201f1e;
      }
      @media screen and (-ms-high-contrast: active){&:hover {
        border-color: Highlight;
        color: Highlight;
      }
      &:active {
        background-color: #edebe9;
        border-color: #8a8886;
        color: #201f1e;
      }
      @media screen and (-ms-high-contrast: active){&:active {
        border-color: Highlight;
        color: Highlight;
      }
      &:hover .ms-Button-icon {
        color: #201f1e;
      }
      @media screen and (-ms-high-contrast: active){&:hover .ms-Button-icon {
        color: Highlight;
      }
      &:active .ms-Button-icon {
        color: #201f1e;
      }
      @media screen and (-ms-high-contrast: active){&:active .ms-Button-icon {
        color: Highlight;
      }
      .ms-Fabric--isFocusVisible &:focus {
        border-style: solid;
      }
  onClick={[Function]}
  role="button"
  tabIndex={0}
  type="button"
>
  <span
    className=
        ms-Stack
        {
          align-items: center;
          box-sizing: border-box;
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          height: 100%;
          justify-content: center;
          padding-bottom: 0px;
          padding-left: 20px;
          padding-right: 20px;
          padding-top: 0px;
          width: auto;
        }
        & > * {
          text-overflow: ellipsis;
        }
        & > *:not(:first-child) {
          margin-left: 8px;
        }
        & > *:not(.ms-StackItem) {
          flex-shrink: 1;
        }
  >
    <span
      className=

          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: inherit;
            display: inline;
            font-family: inherit;
            font-size: 14px;
            font-weight: 600;
            moz-osx-font-smoothing: grayscale;
            overflow: visible;
            webkit-font-smoothing: antialiased;
          }
    >
      Default button
    </span>
  </span>
</button>
`;

exports[`Button renders a disabled primary Button with content correctly 1`] = `
<button
  aria-disabled={true}
  className=
      ms-Button
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background-color: #f3f2f1;
        border-color: #f3f2f1;
        border-radius: 2px;
        border-style: solid;
        border-width: 0px;
        box-sizing: border-box;
        color: #a19f9d;
        cursor: default;
        display: inline-block;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 600;
        justify-content: center;
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        min-height: 32px;
        min-width: 100px;
        outline-color: transparent;
        outline: transparent;
        overflow: hidden;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
        position: relative;
        text-align: center;
        text-decoration: none;
        user-select: none;
        vertical-align: baseline;
      }
      &::-moz-focus-inner {
        border: 0;
      }
      .ms-Fabric--isFocusVisible &:focus:after {
        border: 1px solid #ffffff;
        bottom: 2px;
        content: "";
        left: 2px;
        outline: 1px solid transparent;
        position: absolute;
        right: 2px;
        top: 2px;
        z-index: 1;
      }
      @media screen and (-ms-high-contrast: active){& {
        -ms-high-contrast-adjust: none;
        background-color: Window;
        border-color: GrayText;
        border-width: 1px;
        color: GrayText;
      }
      &:hover {
        background-color: #f3f2f1;
        border-color: #f3f2f1;
        color: #a19f9d;
      }
      @media screen and (-ms-high-contrast: active){&:hover {
        background-color: Window;
        border-color: GrayText;
        color: GrayText;
      }
      &:active {
        background-color: #f3f2f1;
        border-color: #f3f2f1;
        color: #a19f9d;
      }
      @media screen and (-ms-high-contrast: active){&:active {
        background-color: Window;
        border-color: GrayText;
        color: GrayText;
      }
      &:hover .ms-Button-icon {
        color: #a19f9d;
      }
      @media screen and (-ms-high-contrast: active){&:hover .ms-Button-icon {
        color: GrayText;
      }
      &:active .ms-Button-icon {
        color: #a19f9d;
      }
      @media screen and (-ms-high-contrast: active){&:active .ms-Button-icon {
        color: GrayText;
      }
      .ms-Fabric--isFocusVisible &:focus {
        border-style: solid;
        outline-color: transparent;
      }
  disabled={true}
  onClick={[Function]}
  role="button"
  type="button"
>
  <span
    className=
        ms-Stack
        {
          align-items: center;
          box-sizing: border-box;
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          height: 100%;
          justify-content: center;
          padding-bottom: 0px;
          padding-left: 20px;
          padding-right: 20px;
          padding-top: 0px;
          width: auto;
        }
        & > * {
          text-overflow: ellipsis;
        }
        & > *:not(:first-child) {
          margin-left: 8px;
        }
        & > *:not(.ms-StackItem) {
          flex-shrink: 1;
        }
  >
    <span
      className=

          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: inherit;
            display: inline;
            font-family: inherit;
            font-size: 14px;
            font-weight: 600;
            moz-osx-font-smoothing: grayscale;
            overflow: visible;
            webkit-font-smoothing: antialiased;
          }
    >
      Disabled primary button
    </span>
  </span>
</button>
`;

exports[`Button renders a primary Button with content correctly 1`] = `
<button
  className=
      ms-Button
      {
        -moz-osx-font-smoothing: grayscale;
        -webkit-font-smoothing: antialiased;
        background-color: #0078d4;
        border-color: #8a8886;
        border-radius: 2px;
        border-style: solid;
        border-width: 0px;
        box-sizing: border-box;
        color: #ffffff;
        cursor: pointer;
        display: inline-block;
        font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        font-size: 14px;
        font-weight: 600;
        justify-content: center;
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        min-height: 32px;
        min-width: 100px;
        outline-color: transparent;
        outline: transparent;
        overflow: hidden;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
        position: relative;
        text-align: center;
        text-decoration: none;
        user-select: none;
        vertical-align: baseline;
      }
      &::-moz-focus-inner {
        border: 0;
      }
      .ms-Fabric--isFocusVisible &:focus:after {
        border: 1px solid #ffffff;
        bottom: 2px;
        content: "";
        left: 2px;
        outline: 1px solid transparent;
        position: absolute;
        right: 2px;
        top: 2px;
        z-index: 1;
      }
      @media screen and (-ms-high-contrast: active){& {
        -ms-high-contrast-adjust: none;
        background-color: WindowText;
        border-width: 1px;
        color: Window;
      }
      &:hover {
        background-color: #106ebe;
        border-color: #8a8886;
        color: #ffffff;
      }
      @media screen and (-ms-high-contrast: active){&:hover {
        background-color: Highlight;
        border-color: Highlight;
        color: Window;
      }
      &:active {
        background-color: #005a9e;
        border-color: #8a8886;
        color: #ffffff;
      }
      @media screen and (-ms-high-contrast: active){&:active {
        background-color: Highlight;
        border-color: Highlight;
        color: Window;
      }
      &:hover .ms-Button-icon {
        color: #ffffff;
      }
      @media screen and (-ms-high-contrast: active){&:hover .ms-Button-icon {
        color: Window;
      }
      &:active .ms-Button-icon {
        color: #ffffff;
      }
      @media screen and (-ms-high-contrast: active){&:active .ms-Button-icon {
        color: Window;
      }
      .ms-Fabric--isFocusVisible &:focus {
        border-style: solid;
        outline-color: transparent;
      }
  onClick={[Function]}
  role="button"
  tabIndex={0}
  type="button"
>
  <span
    className=
        ms-Stack
        {
          align-items: center;
          box-sizing: border-box;
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          height: 100%;
          justify-content: center;
          padding-bottom: 0px;
          padding-left: 20px;
          padding-right: 20px;
          padding-top: 0px;
          width: auto;
        }
        & > * {
          text-overflow: ellipsis;
        }
        & > *:not(:first-child) {
          margin-left: 8px;
        }
        & > *:not(.ms-StackItem) {
          flex-shrink: 1;
        }
  >
    <span
      className=

          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: inherit;
            display: inline;
            font-family: inherit;
            font-size: 14px;
            font-weight: 600;
            moz-osx-font-smoothing: grayscale;
            overflow: visible;
            webkit-font-smoothing: antialiased;
          }
    >
      Primary button
    </span>
  </span>
</button>
`;
